dolphin-dnn/src/main/java/edu/snu/dolphin/dnn/conf/NeuralNetworkConfigurationBuilder.java

```diff
@@ -35,25 +37,18 @@
  */
 public final class NeuralNetworkConfigurationBuilder implements Builder<Configuration> {
 
-  private Collection<String> layerConfigurations = new ArrayList<>();
-  private int index = 0;
+  private ArrayList<Configuration> layerConfigurations = new ArrayList<>();
```

Comment:
At `build()`, we get the i-th configuration, using `layerConfiguartions.get()`. If we declare this variable as a `List`, the performance `get()` function may be O(n). However, declaring it as an `ArrayList` guarantees O(1) performance. Although it may be the very little performance gain,
This seems to be fine since it is a private member variable,